server/auctionapp/src/test/java/com/atlantbh/auctionapp/service/BidServiceUnitTest.java

```diff
@@ -0,0 +1,153 @@
+package com.atlantbh.auctionapp.service;
+
+import com.atlantbh.auctionapp.exceptions.BadRequestException;
+import com.atlantbh.auctionapp.exceptions.NotFoundException;
+import com.atlantbh.auctionapp.model.BidsEntity;
+import com.atlantbh.auctionapp.model.CategoryEntity;
+import com.atlantbh.auctionapp.model.ProductEntity;
+import com.atlantbh.auctionapp.model.UserEntity;
+import com.atlantbh.auctionapp.repository.BidRepository;
+import com.atlantbh.auctionapp.repository.ProductRepository;
+import com.atlantbh.auctionapp.repository.UserRepository;
+import com.atlantbh.auctionapp.request.BidRequest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+@ExtendWith(MockitoExtension.class)
+public class BidServiceUnitTest {
+
+    @Mock
+    private BidRepository bidRepository;
+    @Mock
+    private ProductRepository productRepository;
+    @Mock
+    private UserRepository userRepository;
+
+    @Test
+    @DisplayName("Return NotFoundException when bids for product are not found")
+    void expectNotFoundExceptionForNonExistingBidsForProductId() {
+        BidService bidService = new BidService(bidRepository, null, null);
```

Comment:
Why not provide user and product repository when you have mocks?
If you do that you can declare private BidService bidService; outside test.
and have one setUp method that will be annotated with @Before and inside that have bidService = new BidService(...)

that setup will be executed before each test, and you only write that line once